Can't read ext4 partition...

...While using `disk.GetFileSystem()`

This is the error I get:
```
DISK: /dev/nvme0n1 PART: 8
DEBU[0000] initDisk(): start
DEBU[0000] initDisk(): block device
DEBU[0000] initDisk(): logical block size 512, physical block size 512
DEBU[0000] trying fat32
DEBU[0000] fat32 failed: error reading MS-DOS Boot Sector: could not read FAT32 BIOS Parameter Block from boot sector: could not read embedded DOS 3.31 BPB: error reading embedded DOS 2.0 BPB: invalid sector size 0 provided in DOS 2.0 BPB. Must be 512
DEBU[0000] trying iso9660 with physical block size 512
DEBU[0000] iso9660 failed: blocksize for ISO9660 must be one of 2048, 4096, 8192
DEBU[0000] trying ext4
DEBU[0000] ext4 failed: could not interpret Group Descriptor Table data: error creating group descriptor from bytes: checksum mismatch, passed 0, actual ad2c
2024/06/26 23:40:56 unknown filesystem on partition 8
panic: unknown filesystem on partition 8
```

The partition is freshly reformatted using `mkfs.ext4`, with no additional options.

The go-diskfs version I use is the latest commit `706241449bae6ba526dffef7d35b2e5a5613522b`

The function I use is this one:
```
func readPartition( diskPath string, partIdx int ) error {
	disk, err := diskfs.Open(diskPath)
	if err != nil {
		return err
	}
	defer disk.Close();

	fs, err := disk.GetFilesystem(partIdx)
	if err != nil {
		log.Panic(err)
	}
	files, err := fs.ReadDir("/") // this should list everything
	if err != nil {
	 	log.Panic(err)
	}
	fmt.Println(files)
	return nil
}
```
The partition I'm trying to read is on a nvme SSD, partition 8 on a GPT table.
The function call is as follow:
`readPartition( "/dev/nvme0n1", 8 )`

Also, `tune2fs -l /dev/nvme0n1p8` gives:
```
tune2fs 1.46.5 (30-Dec-2021)
Filesystem volume name:   <none>
Last mounted on:          <not available>
Filesystem UUID:          8fcec880-f105-4923-b429-e6c6eb16ffc2
Filesystem magic number:  0xEF53
Filesystem revision #:    1 (dynamic)
Filesystem features:      has_journal ext_attr resize_inode dir_index filetype extent 64bit flex_bg sparse_super large_file huge_file dir_nlink extra_isize metadata_csum
Filesystem flags:         signed_directory_hash
Default mount options:    user_xattr acl
Filesystem state:         clean
Errors behavior:          Continue
Filesystem OS type:       Linux
Inode count:              1835008
Block count:              7331072
Reserved block count:     366553
Overhead clusters:        159223
Free blocks:              7171843
Free inodes:              1834997
First block:              0
Block size:               4096
Fragment size:            4096
Group descriptor size:    64
Reserved GDT blocks:      1024
Blocks per group:         32768
Fragments per group:      32768
Inodes per group:         8192
Inode blocks per group:   512
Flex block group size:    16
Filesystem created:       Wed Jun 26 23:33:24 2024
Last mount time:          n/a
Last write time:          Wed Jun 26 23:33:24 2024
Mount count:              0
Maximum mount count:      -1
Last checked:             Wed Jun 26 23:33:24 2024
Check interval:           0 (<none>)
Lifetime writes:          4174 kB
Reserved blocks uid:      0 (user root)
Reserved blocks gid:      0 (group root)
First inode:              11
Inode size:              256
Required extra isize:     32
Desired extra isize:      32
Journal inode:            8
Default directory hash:   half_md4
Directory Hash Seed:      acbf9c6b-2ba9-458b-8ecf-b4596b2c8974
Journal backup:           inode blocks
Checksum type:            crc32c
Checksum:                 0x161dbd42
```

I know ext4 support is a fairly recent addition, so maybe I'm running into something new, here...

--- a/filesystem/ext4/ext4.go
+++ b/filesystem/ext4/ext4.go
@@ -639,7 +639,16 @@ func Read(file util.File, size, start, sectorsize int64) (*FileSystem, error) {
 	gdtSize := uint64(sb.groupDescriptorSize) * sb.blockGroupCount()
 
 	gdtBytes := make([]byte, gdtSize)
-	n, err = file.ReadAt(gdtBytes, start+int64(BootSectorSize)+int64(SuperblockSize))
+	// where do we find the GDT?
+	// - if blocksize is 1024, then 1024 padding for BootSector is block 0, 1024 for superblock is block 1
+	//   and then the GDT starts at block 2
+	// - if blocksize is larger than 1024, then 1024 padding for BootSector followed by 1024 for superblock
+	//   is block 0, and then the GDT starts at block 1
+	gdtBlock := 1
+	if sb.blockSize == 1024 {
+		gdtBlock = 2
+	}
+	n, err = file.ReadAt(gdtBytes, start+int64(gdtBlock)*int64(sb.blockSize))
 	if err != nil {
 		return nil, fmt.Errorf("could not read Group Descriptor Table bytes from file: %v", err)
 	}

